### PR TITLE
Add support for Drupal 10

### PR DESCRIPTION
**Pull request by andytson-inviqa**
not enabled by default but can be enabled

One module not yet with any alpha releases or dev branches for Drupal 10:
* drupal/varnish_purge

Modules with alpha/beta releases:
* drupal/hook_event_dispatcher
* drupal/taxonomy_access_fix
* drupal/drupal-extension - now has a release candidate

### CODE REVIEW

**Review by kierenevans — DISMISSED** [review]

**Comment by andytson-inviqa** [comment]
```
 [notice] Starting Drupal installation. This takes a while.

In InfoParserDynamic.php line 68:

  The 'core_version_requirement' key must be present in modules/custom/startu
  p_core/startup_core.info.yml
````

File: src/drupal/application/skeleton/composer.json.twig [hunk 1]
```diff
@@ -90,7 +97,7 @@
     "drupal/roleassign": "^1.0@alpha",
     "drupal/seckit": "1.1",
     "drupal/stage_file_proxy": "^1.0@alpha",
-    "drupal/taxonomy_access_fix": "2.6",
+    "drupal/taxonomy_access_fix": "^3.3 || ~4.0.0@beta",
```
Comment (andytson-inviqa) [inline src/drupal/application/skeleton/composer.json.twig]:
```suggestion
    "drupal/taxonomy_access_fix": "2.6",
```

wrong place changed

File: src/drupal/application/skeleton/composer.json.twig [hunk 2]
```diff
@@ -104,10 +111,16 @@
...
     "friends-of-behat/mink-extension": "^2.7",
-    "behat/mink-goutte-driver": "^1.3",
+    "behat/mink-goutte-driver": "^1.3 || ^2.0",
     "dmore/behat-chrome-extension": "^1.4",
     "dmore/chrome-mink-driver": "^2.9",
-{% if version_compare(@('drupal.major_version'), '9', '>=') %}
+{% if version_compare(@('drupal.major_version'), '10', '>=') %}
+    "drupal/core-dev": "^10.0",
+    "drupal/devel": "^5.1",
+    "drupal/drupal-extension": "~5.0.0@alpha",
```
Comment (andytson-inviqa) [inline src/drupal/application/skeleton/composer.json.twig]:
```suggestion
    "drupal/drupal-extension": "~5.0.0@rc",
```
A release candidate has now been released, so strengthen the stability

**Review by kierenevans — DISMISSED** [review]

File: src/drupal/application/skeleton/composer.json.twig [hunk 3]
```diff
@@ -104,10 +109,16 @@
...
     "friends-of-behat/mink-extension": "^2.7",
-    "behat/mink-goutte-driver": "^1.3",
+    "behat/mink-goutte-driver": "^1.3 || ^2.0",
     "dmore/behat-chrome-extension": "^1.4",
     "dmore/chrome-mink-driver": "^2.9",
-{% if version_compare(@('drupal.major_version'), '9', '>=') %}
+{% if version_compare(@('drupal.major_version'), '10', '>=') %}
+    "drupal/core-dev": "^10.0",
+    "drupal/devel": "^5.1",
+    "drupal/drupal-extension": "~5.0.0@rc",
```
Comment (andytson-inviqa) [inline src/drupal/application/skeleton/composer.json.twig]:
```suggestion
    "drupal/drupal-extension": "^5.0 || ~5.0.0@rc",
```
same here

**Comment by andytson-inviqa** [comment]
while it might not be a full MVP, it's functional and can be updated later

**Review by kierenevans — APPROVED** [review]

**Pull request merged**